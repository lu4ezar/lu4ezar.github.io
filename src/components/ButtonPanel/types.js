// @flow
import type { Mode } from "../../types/mode";
import type { Todo } from "../../generated/graphql";

export type Props = {
  todo: Todo,
  toggle: (id: string) => void,
  deleteTodo: (id: string) => void,
  showTodo: (id: string, Mode) => void,
};
