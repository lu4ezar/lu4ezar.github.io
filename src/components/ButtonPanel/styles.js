import { IconButton as MuiBtn, Toolbar } from "@material-ui/core";
import styled from "styled-components";

export const IconButton = styled(MuiBtn)`
  padding: 0;
`;

export const StToolbar = styled(Toolbar)`
  display: flex;
  flex-wrap: wrap;
  max-width: 25vmin;
`;
