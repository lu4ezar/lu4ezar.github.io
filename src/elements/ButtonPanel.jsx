// @flow
import React from "react";
import { IconButton, Toolbar } from "@material-ui/core";
import {
  RemoveRedEye as ViewIcon,
  Edit as EditIcon,
  CheckBox,
  CheckBoxOutlineBlank,
  Delete as DeleteIcon,
} from "@material-ui/icons";
import type { Id, Todo } from "../types/todo";

type ButtonPanelProps = {
  todo: Todo,
  toggle: (id: Id) => void,
  deleteTodo: (id: Id) => void,
  showTodo: (id: Id, string) => void,
};

const ButtonPanel = ({
  todo,
  toggle,
  deleteTodo,
  showTodo,
}: ButtonPanelProps) => {
  const { id, status } = todo;
  return (
    <Toolbar>
      <IconButton title="View details" onClick={() => showTodo(id, "view")}>
        <ViewIcon />
      </IconButton>
      <IconButton title="Edit" onClick={() => showTodo(id, "edit")}>
        <EditIcon />
      </IconButton>
      <IconButton title="Mark as Completed" onClick={() => toggle(id)}>
        {status === "completed" ? <CheckBox /> : <CheckBoxOutlineBlank />}
      </IconButton>
      <IconButton title="Delete Todo" onClick={() => deleteTodo(id)}>
        <DeleteIcon />
      </IconButton>
    </Toolbar>
  );
};

export default ButtonPanel;
