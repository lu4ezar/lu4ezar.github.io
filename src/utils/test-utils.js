import React from 'react'
import { render as rtlRender } from '@testing-library/react'
import { MockedProvider } from '@apollo/client/testing'
import { initialMock } from '../components/List/__tests__/List.test.jsx'
import { GET_TODO } from "../apollo/queries"
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import reducer from '../redux/reducers'

const mock = [
  initialMock,
  {
    request: {
      query: GET_TODO,
      variables: {
	id: "1"
      }
    },
    result: {
      data: {
	todo: {
	  id: "1",
	  title: "todo1",
	  description: "desc",
	  status: "ACTIVE",
	  priority: "NORMAL",
	  created: "today"
	}
      }
    }
  }
];

function render(
  ui,
  {
    initialState,
    store = createStore(reducer, initialState),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return(
	    <MockedProvider mocks={mock} addTypename={false}>
	    <Provider store={store}>{children}</Provider></MockedProvider>)
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
export * from '@testing-library/react'
// override render method
export { render }
