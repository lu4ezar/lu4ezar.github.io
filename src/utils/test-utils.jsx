import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { MockedProvider } from "@apollo/client/testing";
import { GraphQLError } from "graphql";
import { GET_TODOS, GET_TODO, GET_ALL } from "../apollo/queries";
import { PriorityValues } from "../generated/graphql";
import type { Todo } from "../generated/graphql";
// import { TOGGLE_TODO } from "../apollo/mutations/todo";

export const lowPriorTodo: Todo = {
  __typename: "Todo",
  id: "1",
  title: "hope",
  description: "wisdom",
  completed: false,
  priority: PriorityValues.Low,
  created: "yesterday",
};

export const normalPriorTodo: Todo = {
  __typename: "Todo",
  id: "2",
  title: "bliss",
  description: "glory",
  completed: false,
  priority: PriorityValues.Normal,
  created: "yesterday",
};

export const normalPriorChecklist: Checklist = {
  __typename: "Checklist",
  id: "3",
  title: "blister",
  description: "inglorious",
  completed: false,
  priority: PriorityValues.Normal,
  created: "yesterday",
};

export const mocks = [
  {
    request: {
      query: GET_TODOS,
    },
    result: {
      data: {
        todos: [lowPriorTodo, normalPriorTodo],
      },
    },
  },
  // {
  //   request: {
  //     query: GET_CHECKLISTS,
  //   },
  //   result: {
  //     data: {
  //       checklists: [normalPriorChecklist],
  //     },
  //   },
  // },
  {
    request: {
      query: GET_ALL,
    },
    result: {
      data: {
        todos: [lowPriorTodo, normalPriorTodo],
        checklists: [normalPriorChecklist],
      },
    },
  },
  {
    request: {
      query: GET_TODO,
      variables: {
        id: "1",
      },
    },
    result: {
      data: {
        todo: normalPriorTodo,
      },
    },
  },
  {
    request: {
      query: GET_TODO,
      variables: {
        id: null,
      },
    },
    result: {
      data: {
        todo: null,
      },
    },
  },
  {
    request: {
      query: GET_ALL,
    },
    error: new GraphQLError("Error!"),
  },
  // {
  //   request: {
  //     mutation: TOGGLE_TODO,
  //     variables: {
  //       id: normalPriorTodo.id,
  //     },
  //   },
  //   result: {
  //     data: {
  //       todo: normalPriorTodo,
  //     },
  //   },
  // },
];

export const emptyMock = [
  {
    request: {
      query: GET_TODOS,
    },
    result: {
      data: {
        todos: [],
      },
    },
  },
];

export const errorMock = [
  {
    request: {
      query: GET_TODOS,
    },
    error: new GraphQLError("Error!"),
  },
];

export const undefinedMock = [
  {
    request: {
      query: GET_TODOS,
    },
    result: { data: undefined },
  },
];

function render(ui, { initialState, ...renderOptions } = {}) {
  function Wrapper({ children }: { children: React.Node }) {
    return (
      <MockedProvider
        mocks={mocks}
        addTypename={false}
        defaultOptions={{
          watchQuery: { fetchPolicy: "no-cache" },
          query: { fetchPolicy: "no-cache" },
        }}
      >
        {children}
      </MockedProvider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from "@testing-library/react";
// override render method
export { render };
