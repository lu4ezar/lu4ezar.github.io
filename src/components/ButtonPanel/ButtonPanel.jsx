// @flow
import * as React from "react";
import {
  RemoveRedEye as ViewIcon,
  Edit as EditIcon,
  CheckBox,
  CheckBoxOutlineBlank,
  Delete as DeleteIcon,
} from "@material-ui/icons";
import { useToggle, useDeleteTodo } from "../../apollo/hooks";
import { StToolbar, IconButton } from "./styles";
import type { Props } from "./types";

const ButtonPanel = ({
  todo: { id, completed },
  showTodo,
}: Props): React.Node => {
  const { deleteTodo } = useDeleteTodo(id);
  const { toggleTodo } = useToggle(id);
  return (
    <StToolbar>
      <IconButton title="View details" onClick={() => showTodo(id, "view")}>
        <ViewIcon />
      </IconButton>
      <IconButton title="Edit" onClick={() => showTodo(id, "edit")}>
        <EditIcon />
      </IconButton>
      <IconButton title="Mark as Completed" onClick={toggleTodo}>
        {completed ? <CheckBox /> : <CheckBoxOutlineBlank />}
      </IconButton>
      <IconButton title="Delete Todo" onClick={deleteTodo}>
        <DeleteIcon />
      </IconButton>
    </StToolbar>
  );
};

export default ButtonPanel;
