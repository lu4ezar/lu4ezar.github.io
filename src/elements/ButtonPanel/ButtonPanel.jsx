// @flow
import React from "react";
import {
  RemoveRedEye as ViewIcon,
  Edit as EditIcon,
  CheckBox,
  CheckBoxOutlineBlank,
  Delete as DeleteIcon,
} from "@material-ui/icons";
import { StToolbar, IconButton } from "./styles";
import type { ButtonPanelProps } from "./types";

const ButtonPanel = ({
  todo: {
    id, status
  },
  toggle,
  deleteTodo,
  showTodo,
}: ButtonPanelProps) => (
    <StToolbar>
      <IconButton title="View details" onClick={() => showTodo(id, "view")}>
        <ViewIcon />
      </IconButton>
      <IconButton title="Edit" onClick={() => showTodo(id, "edit")}>
        <EditIcon />
      </IconButton>
      <IconButton title="Mark as Completed" onClick={() => toggle(id)}>
        {status === "completed" ? <CheckBox /> : <CheckBoxOutlineBlank />}
      </IconButton>
      <IconButton title="Delete Todo" onClick={() => deleteTodo(id)}>
        <DeleteIcon />
      </IconButton>
    </StToolbar>
  );

export default ButtonPanel;
