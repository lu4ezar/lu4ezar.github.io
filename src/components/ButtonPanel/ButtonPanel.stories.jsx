
import React from 'react';
import ButtonPanel from './ButtonPanel';

export default {
  component: ButtonPanel,
  title: 'ButtonPanel',
};

const Template = args => <ButtonPanel {...args} />;

export const Default = Template.bind({});
Default.args = {
  todo: {
    id: '1',
    status: 'ACTIVE',
  },
  // toggle,
  // deleteTodo,
  // showTodo
};

export const Completed = Template.bind({});
Completed.args = {
    ...Default.args,
    todo: {
      ...Default.args.todo,
    status: 'COMPLETED'
    }
  };
