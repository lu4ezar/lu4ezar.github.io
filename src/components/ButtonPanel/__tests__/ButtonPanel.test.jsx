/* eslint-disable react/jsx-props-no-spreading */
import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { MockedProvider } from "@apollo/client/testing";
import ButtonPanel from "../ButtonPanel";
import { mocks } from "../../../utils/test-utils";

const props = {
  todo: {
    id: 0,
    status: "",
  },
  showTodo: jest.fn(),
  edit: jest.fn(),
  toggle: jest.fn(),
};

const renderComponent = () =>
  render(
    <MockedProvider mocks={mocks} addTypename={false}>
      <ButtonPanel {...props} />
    </MockedProvider>
  );

describe("Button Panel", () => {
  it("renders as expected", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
  it("calls showTodo prop function", () => {
    renderComponent();
    userEvent.click(screen.getByTitle(/view/i));
    expect(props.showTodo).toBeCalledWith(props.todo.id, "view");
  });

  it("calls editTodo prop function", () => {
    renderComponent();
    userEvent.click(screen.getByTitle(/edit/i));
    expect(props.showTodo).toBeCalledWith(props.todo.id, "edit");
  });

  it("calls toggleTodo prop function", () => {
    renderComponent();
    userEvent.click(screen.getByTitle(/completed/i));
    expect(props.toggle).toBeCalled();
  });

  xit("calls deleteTodo function", () => {
    renderComponent();
    const deleteTodo = jest.fn();
    userEvent.click(screen.getByTitle(/delete/i));
    expect(deleteTodo).toBeCalled();
  });
});
