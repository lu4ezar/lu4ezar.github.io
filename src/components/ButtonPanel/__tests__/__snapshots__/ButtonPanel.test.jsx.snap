// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Panel renders as expected 1`] = `
<div>
  <div
    class="MuiToolbar-root MuiToolbar-regular sc-gsTCUz TOrPC MuiToolbar-gutters"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root sc-bdfBwQ kBsyEI"
      tabindex="0"
      title="View Details"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          fill="currentColor"
          height="24"
          viewBox="0 0 256 256"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="256"
            width="256"
          />
          <path
            d="M247.31055,124.75061c-.35157-.79-8.81934-19.57617-27.65332-38.41113C194.57324,61.25256,162.87793,47.99182,128,47.99182S61.42676,61.25256,36.34277,86.33948c-18.834,18.835-27.30175,37.62109-27.65332,38.41113a8.00282,8.00282,0,0,0,0,6.49805c.35157.791,8.82032,19.57226,27.6543,38.40429C61.42773,194.734,93.12207,207.99182,128,207.99182S194.57227,194.734,219.65625,169.653c18.834-18.832,27.30273-37.61328,27.6543-38.40429A8.00282,8.00282,0,0,0,247.31055,124.75061ZM128,92.00061a36,36,0,1,1-36,36A36.04061,36.04061,0,0,1,128,92.00061Z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiIconButton-root sc-bdfBwQ kBsyEI"
      tabindex="0"
      title="Edit Todo"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiIconButton-root sc-bdfBwQ kBsyEI"
      tabindex="0"
      title="Mark as Completed"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiIconButton-root sc-bdfBwQ kBsyEI"
      tabindex="0"
      title="Delete Todo"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`Button Panel renders checked checkbox button when todo has 'completed' status 1`] = `
<div>
  <div
    class="MuiToolbar-root MuiToolbar-regular sc-gsTCUz TOrPC MuiToolbar-gutters"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root sc-bdfBwQ kBsyEI"
      tabindex="0"
      title="View Details"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiIconButton-root sc-bdfBwQ kBsyEI"
      tabindex="0"
      title="Edit Todo"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiIconButton-root sc-bdfBwQ kBsyEI"
      tabindex="0"
      title="Mark as Active"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiIconButton-root sc-bdfBwQ kBsyEI"
      tabindex="0"
      title="Delete Todo"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`Button Panel renders empty checkbox button when todo has 'active' status 1`] = `
<div>
  <div
    class="MuiToolbar-root MuiToolbar-regular sc-gsTCUz TOrPC MuiToolbar-gutters"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root sc-bdfBwQ kBsyEI"
      tabindex="0"
      title="View Details"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiIconButton-root sc-bdfBwQ kBsyEI"
      tabindex="0"
      title="Edit Todo"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          fill="currentColor"
          height="24"
          viewBox="0 0 256 256"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="256"
            width="256"
          />
          <path
            d="M224,76.68652,179.314,32a15.99948,15.99948,0,0,0-22.62744,0L130.34473,58.3418l-.00147.00146-.00146.00147L36.687,151.99951A15.89429,15.89429,0,0,0,32,163.31348V208a16.01833,16.01833,0,0,0,16,16H92.68652A15.89438,15.89438,0,0,0,104,219.31348l120.00049-120A16.01823,16.01823,0,0,0,224,76.68652Zm-32,32L147.31372,64,168,43.31348,212.68652,88Z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiIconButton-root sc-bdfBwQ kBsyEI"
      tabindex="0"
      title="Expand todos list"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          fill="currentColor"
          height="24"
          viewBox="0 0 256 256"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="256"
            width="256"
          />
          <path
            d="M215.39111,92.93848A8.00015,8.00015,0,0,0,208,88H48a8,8,0,0,0-5.65674,13.657l80,79.99976a7.99945,7.99945,0,0,0,11.31348,0l80-79.99976A8.00011,8.00011,0,0,0,215.39111,92.93848Z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiIconButton-root sc-bdfBwQ kBsyEI"
      tabindex="0"
      title="Delete Todo"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          fill="currentColor"
          height="24"
          viewBox="0 0 256 256"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="256"
            width="256"
          />
          <g>
            <path
              d="M88,32h80a8,8,0,0,0,0-16H88a8,8,0,0,0,0,16Z"
            />
            <path
              d="M216,48H40a8,8,0,0,0,0,16h8V208a16.01582,16.01582,0,0,0,16,16H192a16.01582,16.01582,0,0,0,16-16V64h8a8,8,0,0,0,0-16ZM112,168a8,8,0,0,1-16,0V104a8,8,0,0,1,16,0Zm48,0a8,8,0,0,1-16,0V104a8,8,0,0,1,16,0Z"
            />
          </g>
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
